lib/dockerregistry/checks.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inventory
+
+import (
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strings"
+
+	gogit "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+// ImageRemovalCheck implements the PreCheck interface and checks against
+// pull requests that attempt to remove any images from the promoter manifests.
+type ImageRemovalCheck struct {
+	GitRepoPath    string
+	MasterSHA      plumbing.Hash
+	PullRequestSHA plumbing.Hash
+}
+
+func isValidSha(potenitalSHA string) error {
+	sha256Length := 64
+	if len(potenitalSHA) == sha256Length {
+		_, err := hex.DecodeString(potenitalSHA)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	return fmt.Errorf("Length of SHA is %d characters, should be 64",
+		len(potenitalSHA))
+}
+
+// MKRealImageRemovalCheck returns an instance of ImageRemovalCheck.
+func MKRealImageRemovalCheck(
+	gitRepoPath string,
+) (*ImageRemovalCheck, error) {
+	masterSHAStr := os.Getenv("PULL_BASE_SHA")
```

Comment:
So there are 2 types of SHAs in our problem domain.

One is the digests for the images. These are sha256. The other is a commit SHA, which is a 40-character string, used by Git. You're checking for Git SHAs to have a 64-char string here (wrong SHA type).

FYI there are many kinds of SHA algorithms. See the wikipedia article for an overview of SHA-1, SHA-2, SHA-3...